docs/site/Inside-Loopback-Application.md

```diff
@@ -0,0 +1,173 @@
+
+- `MongoDB` is used for saving `User` and `Order` data. This demonstrates CRUD operations on a model.
+
+- `Redis` is used for caching items in the shopping cart. This allows demonstrating use of a KV Connector as well as having multiple datasources.
+
+## API Implementation
+
+The API business logic is separated between various layers in LoopBack:
+
+- [Controllers](https://loopback.io/doc/en/lb4/Controllers.html) represent the `API Endpoints`. These endpoints will have to authenticate incoming request, parse and validate as well as orchestrate calls to `Services` and `Repositories`.
+
+- [Services](https://loopback.io/doc/en/lb4/Calling-other-APIs-and-web-services.html) provide common interfaces for external apis and services. This allows invoking external services without mentioning connection details every time. `Services` interchangeably also provide common interfaces for locally available classes.
+
+- [Models](https://loopback.io/doc/en/lb4/Model.html) and [Relations](https://loopback.io/doc/en/lb4/Relations.html) represent domain objects and provide entity relationship models.
+
+- [Repositories](https://loopback.io/doc/en/lb4/Repositories.html) represent the `Entity layer` for a specific model and handle all CRUD operations on the model. They also use repository of other models to handle `entity relations`.
+
+![Business Logic](imgs/shopping-business-logic.png)
+
+## Exposing the API endpoints
+
+> LoopBack develepers can configure simple controller classes and methods as remote endpoints. Out-of-the-box parsing and validation is done using the provided specification.
+
+Developers typically create server-side "handler functions" for each route as remote methods. In LoopBack, remote methods are defined in [controller classes](https://loopback.io/doc/en/lb4/Controllers.html#routing-to-controllers) and configured as API endpoints with api `decorators`.
+
+> LoopBack has built-in decorators to indicate api specification and the expected arguments for the remote method.
+
+For Example,
+- `login()` method in the [UserController](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts) class is defined as `/users/login` API endpoint.
+-  argument of method `findById` in the [UserController](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts) is decorated with `@param.path.string('userId')` which means that the userId parameter in the URL path is passed into the method at runtime.
+
+## API Security
+
+> Security implementations in LoopBack can be created as separate [Authentication strategies](https://loopback.io/doc/en/lb4/Loopback-component-authentication.html) and the `authenticate` decorator can be used to define the authentication strategy of a particular endpoint.
```

Comment:
maybe `@authenticate` instead of `authenticate`?

Reply:
agreed